AllisonsCatPalaceTDDTests/LocationControllerTests.swift

```diff
@@ -0,0 +1,332 @@
+        XCTAssertFalse(controller.activityIndicator.isAnimating,
+                       "Activity indicator should stop animating on beginning text editing")
+    }
+
+    func testZipCodeFieldEndsEditingAtFiveCharacters() {
+        UITextField.ResignFirstResponderSpyController.createSpy(on: textField)?.spy {
+            // Need to be in the window to becomeFirstResponder
+            replaceRootViewController(with: controller)
+            textField.becomeFirstResponder()
+            XCTAssertTrue(textField.isEditing,
+                          "Textfield should be editing when it's the first responder")
+
+            textField.text = "80220"
+            controller.textFieldDidChange(textField)
+
+            XCTAssertFalse(textField.isEditing,
+                           "Textfield should not be editing when it is not the first responder")
+            XCTAssertTrue(textField.resignFirstResponderCalled,
+                          "Textfield should resign first responder at five characters")
+        }
+    }
+
+    func testZipCodeFieldResignsFirstResponderWhenDoneEditing() {
+        UITextField.ResignFirstResponderSpyController.createSpy(on: textField)?.spy {
+            textField.text = "80220"
+            delegate.textFieldDidEndEditing!(textField)
+
+            XCTAssertTrue(textField.resignFirstResponderCalled,
+                          "Textfield should resign first responder when done editing")
+        }
+    }
+
+    func testZipCodeFieldGeocodesAtFiveCharacters() {
+        textField.text = "80220"
+        delegate.textFieldDidEndEditing!(textField)
+        XCTAssertTrue(geocoder.forwardGeocodeAddressCalled)
+    }
+
+    func testActivityIndicatorHiddenByDefault() {
+        XCTAssertTrue(controller.activityIndicator.isHidden,
+                      "Activity indicator should be hidden by default")
+        XCTAssertFalse(controller.activityIndicator.isAnimating,
+                       "Activity indicator should not be animating by default")
+    }
+
+    func testGeocodingInProgress() {
+        textField.text = "80220"
+        delegate.textFieldDidEndEditing!(textField)
+
+        XCTAssertFalse(controller.activityIndicator.isHidden,
+                       "Activity indicator should not be hidden while geocoding")
+        XCTAssertTrue(controller.activityIndicator.isAnimating,
+                      "Activity indicator should animate while geocoding")
+
+        XCTAssertFalse(textField.isUserInteractionEnabled,
+                       "User interaction should be disabled when geocoding is in progress")
+    }
+
+    func testGeocodingCompleted() {
+        textField.text = "80220"
+        delegate.textFieldDidEndEditing!(textField)
+
+        guard let handler = geocoder.forwardGeocodeAddressCompletionHandler else {
+            return XCTFail("Geocoder should be called with a handler")
+        }
+        handler([], nil)
+
+        XCTAssertTrue(controller.activityIndicator.isHidden,
```

Comment:
just make sure that it's stopped since an earlier test checks hides when stopped.